crates/language_models/src/provider/lmstudio.rs

```diff
@@ -531,7 +531,11 @@ impl LmStudioEventMapper {
 
         if let Some(tool_calls) = choice.delta.tool_calls {
             for tool_call in tool_calls {
-                let entry = self.tool_calls_by_index.entry(tool_call.index).or_default();
+                let mut index = tool_call.index.unwrap_or_default();
+                if index < 0 {
+                    index = 0;
```

Comment:
Why are we defaulting to 0 here? 
I would expect -1 to either mean it doesn't match or it is the last one, not the first. If you could provide the docs where this is the case that would help.

But looking at the vllm example you posted, it seems this was just a helper to force logging after certain runs of the loop (an odd way to do it, but I don't think the intention was to actually have a -1 index)

It does feel like it might be better to report the issue to z.ai, we've had a fair number of spec incompatibilities for that provider it seems.